### PR TITLE
Configurable mounting point

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Please advice on the config option names and nesting :cry: 

**Checklist**
- [x] https://github.com/RHEnVision/provisioning-backend#contributing
- [x] https://github.com/RedHatInsights/secure-coding-checklist

### CODE REVIEW

File: cmd/pbapi/main.go [hunk 1]
```diff
@@ -64,11 +64,11 @@ func main() {
 	// Routes for metrics
 	mr := chi.NewRouter()
 	mr.Get("/", statusOk)
-	mr.Handle("/metrics", promhttp.Handler())
+	mr.Handle(config.Prometheus.Path, promhttp.Handler())
 
-	log.Info().Int("webPort", 8000).Int("metricsPort", config.Prometheus.Port).Msg("Starting new instance")
+	log.Info().Int("webPort", config.Application.Port).Int("metricsPort", config.Prometheus.Port).Msg("Starting new instance")
```
Comment (lzap) [inline cmd/pbapi/main.go:69]: Could you actually drop these ints from the logging message here? Or explain what is the benefit. Structured logs are good for searching (give me all logs for a particular request, account, user, organization etc).

File: internal/config/config.go [hunk 2]
```diff
@@ -58,12 +65,16 @@ var Features = &config.FeatureFlags
 func init() {
 	var err error
 
+	viper.SetDefault("app.name", "provisioning")
+	viper.SetDefault("app.pathPrefix", "api")
```
Comment (lzap) [inline internal/config/config.go:69]: I find this a bit unreadable, would you mind refactoring this to just app.prefix being simply "/api/provisioning"? Ideally this value would have a validator that it is actually an absolute path.

**Review by lzap — CHANGES_REQUESTED** [review]
Looks good, for better transparency I suggest to go with simply "prefix".

File: internal/config/config.go (hunk 2 above)
Comment (ezr-ondrej) [reply internal/config/config.go:69]: well app name is configured through Clowder, but the prefix (api) is "fixed" so I'd rather go with having static prefix and configure only the name if that would look better

**Review by lzap — APPROVED** [review]
Thanks.

**Pull request merged**